pkg/webui/lib/errors/utils.js

```diff
@@ -0,0 +1,155 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import grpcErrToHttpErr from './grpc-error-map'
+
+/**
+ * Tests wether the error is a backend error object
+ * @param {Object} error - The error to be tested.
+ * @returns {boolean} `true` if `error` is translated, `false` otherwise.
+ */
+export const isBackend = error => (
+  typeof error === 'object'
+    && !('id' in error)
+    && error.message
+    && error.details
+    && (error.code || error.grpc_code)
+)
+
+/**
+ * Returns wether the error is a frontend defined error object
+ * @param {Object} error - The error to be tested.
+ * @returns {boolean} `true` if `error` is translated, `false` otherwise.
+ */
+export const isFrontend = error => (
+  // TODO: Define proper object shape, once we need it, for now translated
+  // messages are enough
+  typeof error === 'object' && error.id && error.defaultMessage
+)
+
+/**
+ * Returns wether the error has a shape that is well-known
```

Comment:
misleading comment imo
according to the comment i would expect the function to return `true` when the error shape is `well-known`